Firestore/core/src/firebase/firestore/public/log.h

```diff
@@ -0,0 +1,42 @@
+/*
```

Comment:
The objective-C code has a Public folder because of warts in the way cocoapods are constructed. We don't need this for C++ code at all.

Let's not have a public folder. Please either put these contents in `util/log.h` or create `util/log_level.h`.

Reply:
firebase::firestore is our public namespace; namespaces nested within there are nonpublic. Based on this I prefer this to be in util.